src/lib_c/x86_64-linux-musl/c/sys/types.cr

```diff
@@ -58,4 +58,9 @@ lib LibC
   alias SusecondsT = Long
   alias TimeT = Long
   alias UidT = UInt
+
+  struct IoVec
+    iov_base : Void*
+    iov_len : SizeT
+  end
```

Comment:
argh, I didn't realize that my suggestion to put it in libc would force it to be duplicated like this. 

@straight-shoota Hmm, do you know if it has been considered to have a common place to put definitions common for the operating systems (linux in this case though the definition would be identical for *bsd/mac) and limit the machine specific stuff to what actually vary by arch and not by OS?

Reply:
Portable lib definitions can simply be placed in a common path like `crystal/system/unix`.

Reply:
Many more structs and constants don't vary on arch or libc implementation, and some are the same across all Unixes. For example: `MAP_*` are equal for all Linux, or `LibC::Group` struct is the same in 19 places. It could be better, but this is an existing problem, not a new one caused by IoVec. I prefer leaving it as is (this is how it has always been done) and refactor for all of libc at once later.